crates/re_log_types/src/size_bytes_ext.rs

```diff
@@ -0,0 +1,90 @@
+use arrow2::datatypes::{DataType, Field};
+
+// ---
+
+/// A trait to implement `size_bytes` method on types we don't own.
+pub trait SizeBytesExt {
+    /// Returns an accurate-enough approximation of the heap allocated size of `self`.
```

Comment:
You should add a note here about how implementors should handle the contents of ref-counted data.

Reply:
You sometimes include `std::mem::size_of_val(self)` in the implementations, and sometimes you don't. You should document here what is intended.

Reply:
You should also document how to handle size vs capacity of e.g. a `Vec`